en/lessons/libraries/bypass.md

```diff
@@ -0,0 +1,274 @@
+---
+version: 1.0.0
+title: Bypass
+redirect_from:
+  - /lessons/libraries/bypass/
+---
+
+When testing our applications there are often times we need to make requests to external services, we may even want to simulate different situations like unexpected server errors.
+Handling this in an efficient way doesn't come easy in Elixir without a little help.  
+
+In this lesson we're going to explore how [bypass](https://github.com/PSPDFKit-labs/bypass) can help us quickly and easily handle these requests in our tests.
+
+{% include toc.html %}
+
+## What is Bypass?
+
+[Bypass](https://github.com/PSPDFKit-labs/bypass) is described as "a quick way to create a custom plug that can be put in place instead of an actual HTTP server to return prebaked responses to client requests."
+
+What does that mean?
+Under-the-hood Bypass is an OTP application that masquerades as an external server listening for and responding to requests.
+By responding with pre-defined responses you can test any number of possibilities such as services being unavailable, unexpected server errors, and any expected errors you might encounter.
```

Comment:
Maybe mention that this approach is favorable because it leans on dependency injection per-environment instead of mocking/stubbing?

Reply:
lol btw this is bobby @doomspork I forgot I was logged in on my alt acct but w/e

Reply:
I figured!  Why two accounts though?!

Reply:
I accidentally got locked out of my personal email and forgot the password and havent been able to reset anything cuz I'm a super-genius.